Tests/tests/test_Warehouse.cpp

```diff
@@ -0,0 +1,803 @@
+            return Core::ERROR_NONE;
+        }))
+    .WillOnce(::testing::Invoke(
+        [](const char* command) {
+            EXPECT_EQ(string(command), string("sh /lib/rdk/deviceReset.sh warehouse"));
+            return Core::ERROR_NONE;
+        }));
+
+    //reset: suppress reboot: false
+    EXPECT_EQ(Core::ERROR_NONE, handler.Invoke(connection, _T("resetDevice"), _T("{\"suppressReboot\":false}"), response));
+    EXPECT_EQ(response, _T("{\"success\":true}"));
+    EXPECT_EQ(Core::ERROR_NONE, resetDone.Lock());
+    //Disable Power manager 2
+    plugin->m_isPwrMgr2RFCEnabled = false;
+}
+
+TEST_F(WarehouseInitializedTest, GenericResetDeviceNoResponsePwrMgr2RFCEnabled)
+{
+    //Enable Power manager 2
+    plugin->m_isPwrMgr2RFCEnabled = true;
+
+    EXPECT_CALL(wrapsImplMock, system(::testing::_))
+    .Times(2)
+    .WillOnce(::testing::Invoke(
+        [](const char* command) {
+            EXPECT_EQ(string(command), string("touch /tmp/.warehouse-reset"));
+            return Core::ERROR_NONE;
+        }))
+    .WillOnce(::testing::Invoke(
+        [](const char* command) {
+            EXPECT_EQ(string(command), string("sh /lib/rdk/deviceReset.sh warehouse --suppressReboot &"));
+            return Core::ERROR_NONE;
+        }));
+
+    //reset: suppress reboot: true - This doesn't generate any event (Expect no response)
+    EXPECT_EQ(Core::ERROR_NONE, handler.Invoke(connection, _T("resetDevice"), _T("{\"suppressReboot\":true}"), response));
+    EXPECT_EQ(response, _T("{\"success\":true}"));
+    usleep(10 * 1000);  //Wait for reset thread to finish
+    //Disable Power manager 2
+    plugin->m_isPwrMgr2RFCEnabled = false;
+}
+
+TEST_F(WarehouseResetDeviceFailureTest, UserFactoryResetDeviceFailurePwrMgr2RFCEnabled)
+{
+    //Enable Power manager 2
+    plugin->m_isPwrMgr2RFCEnabled = true;
+
+    EXPECT_CALL(wrapsImplMock, system(::testing::_))
+    .Times(2)
+    .WillOnce(::testing::Invoke(
+        [](const char* command) {
+            EXPECT_EQ(string(command), string("touch /tmp/.warehouse-reset"));
+            return Core::ERROR_NONE;
+        }))
+    .WillOnce(::testing::Invoke(
+        [](const char* command) {
+            EXPECT_EQ(string(command), string("sh /lib/rdk/deviceReset.sh warehouse --suppressReboot &"));
+            return Core::ERROR_GENERAL;
+        }));
+
+    //reset: suppress reboot: true - This doesn't generate any event (Expect no response)
+    EXPECT_EQ(Core::ERROR_NONE, handler.Invoke(connection, _T("resetDevice"), _T("{\"suppressReboot\":true}"), response));
+    EXPECT_EQ(response, _T("{\"success\":true}"));
+    usleep(10 * 1000);  //Wait for reset thread to finish
+    //Disable Power manager 2
+    plugin->m_isPwrMgr2RFCEnabled = false;
+}
+
+TEST_F(WarehouseInitializedTest, setFrontPanelState)
+{
+#ifndef HAS_FRONT_PANEL
```

Comment:
please remove #ifndef as there is only 1 way of building unit tests

Reply:
Done. In future, when "HAS_FRONT_PANEL" gets enabled, then this test will fail.

Reply:
since API is no-op without HAS_FRONT_PANEL a test for it is not needed. please remove

Reply:
Done